nucypher/crypto/utils.py

```diff
@@ -133,3 +145,166 @@ def get_coordinates_as_bytes(point: Union[Point, UmbralPublicKey, SignatureStamp
         return coordinates_as_bytes[middle:]
     else:
         raise ValueError("At least one coordinate must be set")
+
+
+SYSTEM_RAND = SystemRandom()
+
+
+class InvalidNodeCertificate(RuntimeError):
+    """Raised when an Ursula's certificate is not valid because it is missing the checksum address."""
+
+
+def secure_random(num_bytes: int) -> bytes:
+    """
+    Returns an amount `num_bytes` of data from the OS's random device.
+    If a randomness source isn't found, returns a `NotImplementedError`.
+    In this case, a secure random source most likely doesn't exist and
+    randomness will have to found elsewhere.
+
+    :param num_bytes: Number of bytes to return.
+
+    :return: bytes
+    """
+    # TODO: Should we just use os.urandom or avoid the import w/ this?
+    return SYSTEM_RAND.getrandbits(num_bytes * 8).to_bytes(num_bytes, byteorder='big')
```

Comment:
`os.random` would be a bit more concise. For a reference, `SystemRandom` is using `os.urandom` [underneath](https://github.com/python/cpython/blob/main/Lib/random.py#L800), and both of those methods are not cross-platform compatible.

Reply:
I'm somewhat compelled to use the `secrets` module as a source of randomness for added paranoia. <https://docs.python.org/3/library/secrets.html>

Reply:
Switched to the `secrets` version of `SystemRandom`.